PolyEngine/Engine/Src/Imgui/ImguiSystem.cpp

```diff
@@ -48,118 +121,63 @@ void ImguiSystem::ImguiUpdatePhase(Scene* scene)
 			io.KeySuper = inputCmp->IsPressed(eKey::LGUI) || inputCmp->IsPressed(eKey::RGUI);
 		}
 	}
-		
-	// Set OS mouse position if requested (rarely used, only when ImGuiConfigFlags_NavEnableSetMousePos is enabled by user)
-	// if (io.WantSetMousePos)
-	// 	SDL_WarpMouseInWindow(g_Window, (int)io.MousePos.x, (int)io.MousePos.y);
-	// else
-	//	io.MousePos = ImVec2(-FLT_MAX, -FLT_MAX);
-	io.MousePos = ImVec2(-FLT_MAX, -FLT_MAX);
-
-	// int mx, my;
-	// Uint32 mouse_buttons = SDL_GetMouseState(&mx, &my);
+
+	// used to read keyboard and fill text input Widgets
+	if (inputCmp->GetCharUTF8() != nullptr)
+	{
+		const char* charUTF8 = inputCmp->GetCharUTF8();
+		size_t length = StrLen(charUTF8);
```

Comment:
This is dangerous. The length should be handled by utf-8 aware method. We really need a UTF-8 String... But do we really need to check that length? I would just check if it's nullptr.

Reply:
nullptr was check was added later and should be sufficient, done